src/test/java/de/szut/lf8_project/integration/TestRemoveEmployeeFromProject.java

```diff
@@ -0,0 +1,110 @@
+import de.szut.lf8_project.common.Errorcode;
+import de.szut.lf8_project.domain.employee.Employee;
+import de.szut.lf8_project.domain.employee.ProjectRole;
+import de.szut.lf8_project.domain.project.Project;
+import de.szut.lf8_project.domain.project.TeamMember;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Nested;
+import org.junit.jupiter.api.Test;
+import org.springframework.http.MediaType;
+import org.springframework.test.web.servlet.ResultActions;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.hamcrest.core.Is.is;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.delete;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.jsonPath;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+@DisplayName("Die remove employee from project Methode")
+public class TestRemoveEmployeeFromProject extends FullIntegrationTest {
+
+    @Test
+    @DisplayName("sollte einen Mitarbeiter erfolgreich entfernen")
+    void shouldRemove() throws Exception {
+        Project project = createAndSaveDefaultProjectWithProjectLead();
+        Employee employee = saveEmployeeInRemoteRepository(createDefaultEmployeeWithout0Id());
+        project.getTeamMembers().add(new TeamMember(
+               employee.getId(),
+               new ProjectRole("egal")
+        ));
+        project = saveProjectInDatabase(project);
+
+        ResultActions result = mockMvc.perform(
+                delete("/api/v1/project/" + project.getProjectId().get().unbox() + "/removeEmployee/" + employee.getId().unbox())
+                        .header("Authorization", jwt.jwt())
+                        .contentType(MediaType.APPLICATION_JSON)
+        );
+
+        result.andExpect(status().isOk());
+        assertThat(getProjectByIdFromDatabase(project.getProjectId().get()).getTeamMembers()).isEmpty();
+
+    }
+
+    @Nested
+    @DisplayName("sollte keinen Mitarbeiter entfernen")
+    class TestDontRemoveEmployeeFromProject {
+
+        @Test
+        @DisplayName("wenn das Projekt nicht existiert")
+        void projectNotFound() throws Exception {
+
+            ResultActions result = mockMvc.perform(delete("/api/v1/project/868668/removeEmployee/2323")
+                    .header("Authorization", jwt.jwt())
+                    .contentType(MediaType.APPLICATION_JSON)
+            );
+
+            result
+                    .andExpect(status().isNotFound())
+                    .andExpect(jsonPath("$.title", is(Errorcode.ENTITY_NOT_FOUND.toString())));
+
+        }
+
+        @Test
+        @DisplayName("wenn der Mitarbeiter nicht im Projekt ist")
+        void employeeNotInProject() throws Exception {
+            Project project = createAndSaveDefaultProjectWithProjectLead();
+
+            ResultActions result = mockMvc.perform(delete("/api/v1/project/" + project.getProjectId().get().unbox() + "/removeEmployee/2323")
```

Comment:
Sollte der Employee dann nicht wenigstens in der externen API gespeichert sein, damit der Test aussagekräftig ist?

Reply:
nötig ist es halt rein technisch nicht - ich will ja auch einen mitarbeiter entfernen können, der vll im remote vorher rausgeflogen ist.

ein check auf vorhandensein des employees im remote repo vorm löschen würde uns ja auch irgendwie nicht helfen